### PR TITLE
Overdue document enhancements to explain overdue.xml file

### PR DESCRIPTION
**Pull request by reshmabidikar**

### CODE REVIEW

File: userguide/tutorials/overdue.adoc [hunk 1]
```diff
@@ -512,6 +512,139 @@ Even if the payment retry schedule is not configured explicitly, there is a defa
...
+|autoReevaluationInterval   |Tag within `state`   |Specifies the duration after which the overdue state should be reevaluated. The duration needs to be specified with a `unit` (one of `DAYS`, `WEEKS`, `MONTHS`, `YEARS`, `UNLIMITED`) and an integer value. |  [source, xml]
+                <autoReevaluationInterval>
+                 <unit>DAYS</unit>
+                 <number>4</number>
+                </autoReevaluationInterval>
+|isClearState   |Tag within `state`   |Specifies whether the current state is a clear state. A true/false value needs to be specified. |
+[source, xml]
+ <isClearState>true</isClearState>
+|enterStateEmailNotification   |Tag within `state`   |Deprecated|
+NA
```
Comment (reshmabidikar) [inline userguide/tutorials/overdue.adoc]: Is this correct? I assumed that this was deprecated based on [DefaultOverdueState#Line76](https://github.com/killbill/killbill/blob/2b80b0445c7baf1f613425bb236a8cb36f1f377a/overdue/src/main/java/org/killbill/billing/overdue/config/DefaultOverdueState.java#L76)

File: userguide/tutorials/overdue.adoc [hunk 2]
```diff
@@ -512,6 +512,139 @@ Even if the payment retry schedule is not configured explicitly, there is a defa
...
+[source, xml]
+<condition>
+  <responseForLastFailedPaymentIn>
+    <response>INVALID_CARD</response>
+    <response>LOST_OR_STOLEN_CARD</response>
+     <response>INSUFFICIENT_FUNDS</response>
+  </responseForLastFailedPaymentIn>
+</condition>
+|controlTagInclusion   |Specifies that the overdue flow should kick in when the account is tagged with the specified control tag. Note that this condition cannot be specified by itself and can only be specified along with one of the other conditions as shown. | 
+[source, xml]
```
Comment (reshmabidikar) [inline userguide/tutorials/overdue.adoc]: Again, could you confirm if this is correct? I found that when the `controlTagInclusion` condition is specified by itself, the overdue flow is not triggered. It only works when the  `controlTagInclusion` condition is used along with some other condition.

File: userguide/tutorials/overdue.adoc [hunk 3]
```diff
@@ -512,6 +512,139 @@ Even if the payment retry schedule is not configured explicitly, there is a defa
...
+<subscriptionCancellationPolicy>
+END_OF_TERM
+</subscriptionCancellationPolicy>`
+|blockChanges   |Tag within `state`|Specifies whether the customer is allowed to make any plan change on the subscriptions when this overdue state is reached. A true/false value needs to be specified. |
+[source, xml]
+<blockChanges>
+true
+</blockChanges>
+|disableEntitlementAndChangesBlocked  |Tag within `state`|Specifies whether the subscriptions should be paused (which means entitlement and billing will be disabled) when this overdue state is reached. A true/false value needs to be specified.    |
```
Comment (pierre) [inline userguide/tutorials/overdue.adoc]: This blocks billing and entitlement, and also blocks any changes.

File: userguide/tutorials/overdue.adoc (hunk 1 above)
Comment (pierre) [reply userguide/tutorials/overdue.adoc]: Correct - maybe we don't even mention it?

File: userguide/tutorials/overdue.adoc [hunk 4]
```diff
@@ -512,6 +512,139 @@ Even if the payment retry schedule is not configured explicitly, there is a defa
...
+</condition> 
+|timeSinceEarliestUnpaidInvoiceEqualsOrExceeds   |Specifies that the overdue flow should kick in when the specified duration has elapsed since the last unpaid invoice. The duration needs to be specified with a `unit` (one of `DAYS`, `WEEKS`, `MONTHS`, `YEARS`, `UNLIMITED`) and an integer value  | 
+[source, xml]
+<condition>
+  <timeSinceEarliestUnpaidInvoiceEqualsOrExceeds>
+    <unit>DAYS</unit>
+    <number>10</number>
+  </timeSinceEarliestUnpaidInvoiceEqualsOrExceeds>
+</condition>
+|responseForLastFailedPaymentIn|Specifies that the overdue flow should kick in when payment fails and the failed response is one of the specified values.  Possible values are defined in https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentResponse.java[PaymentResponse]. Multiple values may be specified.    |
```
Comment (pierre) [inline userguide/tutorials/overdue.adoc]: This was actually never implemented :-) https://github.com/killbill/killbill/blob/00d9ace820bc7ddf6961a16a2727abe17b198bf9/overdue/src/main/java/org/killbill/billing/overdue/calculator/BillingStateCalculator.java#L84

You can remove too.

File: userguide/tutorials/overdue.adoc (hunk 2 above)
Comment (pierre) [reply userguide/tutorials/overdue.adoc]: Weird: I would expect it to behave just like any other condition. https://github.com/killbill/killbill/blob/00d9ace820bc7ddf6961a16a2727abe17b198bf9/overdue/src/main/java/org/killbill/billing/overdue/config/DefaultOverdueCondition.java#L69-L84

Not a blocker for this PR but might be worth debugging why it's happening.

**Review by pierre — APPROVED** [review]
The table format is a nice touch 👌

File: userguide/tutorials/overdue.adoc (hunk 1 above)
Comment (reshmabidikar) [reply userguide/tutorials/overdue.adoc]: Fixed

File: userguide/tutorials/overdue.adoc (hunk 2 above)
Comment (reshmabidikar) [reply userguide/tutorials/overdue.adoc]: I've updated the document for now, will investigate this sometime later.

File: userguide/tutorials/overdue.adoc (hunk 3 above)
Comment (reshmabidikar) [reply userguide/tutorials/overdue.adoc]: Updated the description to reflect this

File: userguide/tutorials/overdue.adoc (hunk 4 above)
Comment (reshmabidikar) [reply userguide/tutorials/overdue.adoc]: Fixed

File: userguide/tutorials/overdue.adoc [hunk 5]
```diff
@@ -512,6 +512,124 @@ Even if the payment retry schedule is not configured explicitly, there is a defa
...
+|disableEntitlementAndChangesBlocked  |Tag within `state`|Specifies whether the subscriptions should be paused (which means entitlement and billing will be disabled and changes will be blocked) when this overdue state is reached. A true/false value needs to be specified.    |
+[source, xml]
+<disableEntitlementAndChangesBlocked>
+true
+</disableEntitlementAndChangesBlocked>
+|autoReevaluationInterval   |Tag within `state`   |Specifies the duration after which the overdue state should be reevaluated. The duration needs to be specified with a `unit` (one of `DAYS`, `WEEKS`, `MONTHS`, `YEARS`, `UNLIMITED`) and an integer value. |  [source, xml]
+                <autoReevaluationInterval>
+                 <unit>DAYS</unit>
+                 <number>4</number>
+                </autoReevaluationInterval>
```
Comment (ghost) [inline userguide/tutorials/overdue.adoc:555]: On the GitHub view of the table, this code is aligned pretty far over to the right. Just want to make sure that's what you wanted.

**Review by ghost — APPROVED** [review]

**Pull request merged**